popdat/src/import_census.rs

```diff
@@ -1,11 +1,95 @@
-use map_model::Map;
-
 use crate::CensusArea;
+use abstutil::Timer;
+use geo::algorithm::intersects::Intersects;
+use geojson::GeoJson;
+use map_model::Map;
+use std::convert::TryFrom;
 
 impl CensusArea {
-    pub fn find_data_for_map(_map: &Map) -> Result<Vec<CensusArea>, String> {
-        // TODO importer/src/utils.rs has a download() helper that we could copy here. (And later
-        // dedupe, after deciding how this crate will integrate with the importer)
-        todo!()
+    pub fn find_data_for_map(map: &Map, timer: &mut Timer) -> Result<Vec<CensusArea>, String> {
+        // TODO eventually it'd be nice to lazily download the info needed. For now we expect a
+        // prepared geojson file to exist in data/system/<city>/population_areas.geojson
+        //
+        // When we implement downloading, importer/src/utils.rs has a download() helper that we
+        // could copy here. (And later dedupe, after deciding how this crate will integrate with
+        // the importer)
+        let path = abstutil::path(format!(
+            "system/{}/population_areas.geojson",
+            map.get_name().city
+        ));
+        let bytes = abstutil::slurp_file(&path)?;
+        debug!("parsing geojson at path: {}", &path);
+
+        // TODO - can we change to Result<_,Box<dyn std::error::Error>> and avoid all these map_err?
+        let str = String::from_utf8(bytes).map_err(|e| e.to_string())?;
+        timer.start("parsing geojson");
+        let geojson = str.parse::<GeoJson>().map_err(|e| e.to_string())?;
+        timer.stop("parsing geojson");
+        let mut results = vec![];
+        let collection =
+            geojson::FeatureCollection::try_from(geojson).map_err(|e| e.to_string())?;
+
+        let map_area = map.get_boundary_polygon();
+        let bounds = map.get_gps_bounds();
+
+        use geo::algorithm::map_coords::MapCoordsInplace;
+        let mut geo_map_area: geo::Polygon<_> = map_area.clone().into();
+        geo_map_area.map_coords_inplace(|c| {
+            let projected = geom::Pt2D::new(c.0, c.1).to_gps(bounds);
```

Comment:
Is there a better way to get the map poly in wgs84?

Though maybe it's irrelevant, because the other question I need to answer is: is there a blessed way to convert a poly *from* wgs84 *to* map coords?

Reply:
`GPSBounds::convert`? Or if you want to enforce all points are in bounds, `try_convert`?